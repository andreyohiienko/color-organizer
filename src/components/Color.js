import React, { Component } from 'react'
import PropTypes from 'prop-types'
import StarRating from './StarRating'
import TimeAgo from './TimeAgo'
import { FaTrash } from 'react-icons/fa'
import { rateColor, removeColor } from '../actions'
import '../stylesheets/Color.scss'

class Color extends Component {
  render() {
    const { id, title, color, rating, timestamp } = this.props
    const { store } = this.context
    return (
      <section className="color" style={this.style}>
        <h1 ref="title">{title}</h1>
        <button onClick={() => store.dispatch(removeColor(id))}>
          <FaTrash />
        </button>
        <div className="color" style={{ backgroundColor: color }}></div>
        <TimeAgo timestamp={timestamp}></TimeAgo>
        <div>
          <StarRating
            starsSelected={rating}
            onRate={rating => store.dispatch(rateColor(id, rating))}
          />
        </div>
      </section>
    )
  }
}

Color.contextTypes = {
  store: PropTypes.object,
}

Color.propTypes = {
  title: PropTypes.string,
  rating: PropTypes.number,
  color: PropTypes.string,
  onRate: PropTypes.func,
}

Color.defaultProps = {
  rating: 0,
}
export default Color
