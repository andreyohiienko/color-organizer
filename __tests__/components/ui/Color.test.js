import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Color from '../../../src/components/ui/ColorList'
import { compose } from 'redux'

describe('<Color /> UI Component', () => {
  // const shallowExpect = compose(expect, toJSON, shallow)

  it('Renders correct properties', () => {
    let output = shallow(
      <Color
        title="Test Color"
        color="#f0f0f0"
        rating={3}
        timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)"
      />,
    ).html()

    expect(output).toMatchSnapshot()
  })
})
