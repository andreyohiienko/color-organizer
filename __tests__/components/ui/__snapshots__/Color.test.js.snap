// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Color /> UI Component Renders correct properties 1`] = `"<div class=\\"color-list\\"><p>No Colors listed. (Add a Color)</p></div>"`;
